src/visualization/ColorPalette/ColorMap.cc

```diff
@@ -0,0 +1,284 @@
+#include <string>
+#include <vector>
+#include <SFML/Graphics/Color.hpp>
+#include <array>
+#include <cmath>
+#include "ColorMap.h"
+#include <iostream>
+
+using namespace std;
+using namespace sf;
+
+/**
+ * This Method will compute in terms of the argument sequences and the k, all kmers we are looking for and call
+ * the method which compute us in terms of this number a Colormap. Then the method will compute
+ * all different Kmers and push them in a List. We'll use this list to initialize differentKmers.
+ */
+ColorMap::ColorMap(vector<string>& Sequences, int k){
+    ///< This is a variable which save the number of Kmers per Sequence which we have
+    vector<int> SequenceSizes;
+    for (int i=0;i<Sequences.size();i++)
+        SequenceSizes[i] = (int)floor((Sequences[i].size())/k);
+
+    ///< This is a variable which save the number of Kmers
+    int numbOfKmer = 0;
+    for (int i=0;i<SequenceSizes.size();i++)
+        numbOfKmer += SequenceSizes[i];
+    updateColorList(numbOfKmer);
```

Comment:
It does not make sense to call that function, that determines the colors, now, as not even the number of required colors is determined yet. As equal kmers should be colored equal (to visualize what can be matched), `numbOfKmer` is now generally larger than the required number of colors.